src/main/java/com/puppycrawl/tools/checkstyle/api/Configuration.java

```diff
@@ -30,20 +30,20 @@
 public interface Configuration extends Serializable {
 
     /**
-     * The set of attribute names.
+     * The set of property names.
      *
-     * @return The set of attribute names, never null.
+     * @return The set of property names, never null.
      */
-    String[] getAttributeNames();
+    String[] getPropertyNames();
```

Comment:
this is API, we can not damage it, CI failures are probably is proving this.

Please create new methods `xxxPropertyXxxx` and deprecate  `xxxAttributeXxxx` (xxxAttributeNames will simply call xxxPropertyXxxx in implementation).
Whole checkstyle should migrate to `xxxPropertyXxxx`. Later in future we will remove xxxAttributexxxx.

Reply:
done, please take another look